test/jdk/java/net/httpclient/http2/StreamFlowControlTest.java

```diff
@@ -307,13 +341,20 @@ public void handle(Http2TestExchange t) throws IOException {
                             // ignore and continue...
                         }
                     }
-                    ((BodyOutputStream) os).writeUncontrolled(resp, 0, resp.length);
+                    try {
+                        ((BodyOutputStream) os).writeUncontrolled(resp, 0, resp.length);
+                    } catch (IOException x) {
+                        if (t instanceof FCHttp2TestExchange fct) {
+                            fct.conn.updateConnectionWindow(resp.length);
+                        }
+                    }
                 }
+            } finally {
+                if (t instanceof FCHttp2TestExchange fct) {
+                    fct.responseSent(query);
+                } else fail("Exchange is not %s but %s"
```

Comment:
Nit, while we are here, I think it might be better to wrap the contents of the `else` block in a `{}` for consistency.